heltour/tournament/forms.py

```diff
@@ -36,6 +36,23 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         self.season = kwargs.pop('season')
+        
+        username = kwargs.pop('user')
+        
+        #try to find a SeasonPlayer object for the current season.
+        #if SeasonPlayer object already exists for the current season we know the player got accepted before 
+        #and we do not have to display the availability fields in the registration form.
+        try:
+            #the lonewolf seasons are organized in section groups. Registrations are always linked to the lonewolf open section
+            #but players might be assigned to the u1800 section. Hence we have to identify all seasons in the same section group.
+            #On the other hand, team league and 960 do not have sections and we have to check directly the season.
+            seasonsInGroup = Season.objects.filter(
+                section__in=Section.objects.filter(
+                    section_group=Section.objects.filter(season=self.season).first().section_group))
+        except:
+            seasonsInGroup = Season.objects.filter(name=self.season.name)
+        already_accepted = SeasonPlayer.objects.filter(season__in=seasonsInGroup, player__lichess_username=username).exists()
```

Comment:
I did not find a smarter way to write this. Problem is that for lonewolf the seasons are linked with each other via Section and SectionGroup, but team league and chess960 have no section. Pretty sure there is a better way than try/except but the checks that I have tried (self.season.section is None) resulted in exceptions for the team league.

Reply:
You could use the same `Q` objects we did previously.  What, exactly, are the exceptions that are throw in the first case?

Reply:
I'm unclear why this wouldn't work:
```python
potential_sections = Section.objects.filter(
    season=self.season
)

seasonsInGroup = Season.objects.filter(
    Q(section__in=potential_sections) | Q(name=self.season.name)
)

already_accepted = SeasonPlayer.objects.filter(season__in=seasonsInGroup, player__lichess_username=username).exists()
```

Reply:
Argh, Season has a method that returns exactly what I want (facepalm), see recent commit.